src/integrate/scarches/config.vsh.yaml

```diff
@@ -0,0 +1,140 @@
+functionality:
+  name: scarches
+  namespace: "integrate"
+  version: "dev"
+  description: "Performs reference mapping with scArches"
+  authors:
+    - name: Vladimir Shitov
+      email: vladimir.shitov@helmholtz-muenchen.de
+      roles: [ author ]
+      props: { github: vladimirshitov, orcid: "0000-0002-1960-8812" }
+  argument_groups:
+    - name: Inputs
+      arguments:
+        - name: "--input"
+          alternatives: ["-i"]
+          type: file
+          description: Input h5mu file to use as a query
+          direction: input
+          required: true
+        - name: "--modality"
+          type: string
+          default: "rna"
+          required: false
+        - name: "--reference"
+          choices: ["HLCA"]
+          alternatives: ["-r"]
+          type: string
+          description: Name of the reference dataset
+          required: false
+        - name: "--unlabeled_category"
+          type: string
+          description: "Value used for unlabeled cells"
+          default: "Unknown"
+          required: false
+        - name: "--labels_key"
+          type: string
+          description: "Name of the column containing cell types"
+          default: "leiden"
+          required: false
+        - name: "--predicted_labels_key"
+          type: string
+          description: Name of the column that will contain cell types predicted by scANVI
+          default: "predicted_labels"
+          required: false
+        - name: "--input_layer"
+          type: string
+          required: false
+          description: "Input layer to use. If None, X is used"
+        - name: "--obs_batch"
+          type: string
+          default: "sample_id"
+          required: false
+          description: Column name discriminating between your batches.
```

Comment:
* batch effect key from ref model you can get from the model
* batch effect you want to correct for in query data, you may want to get as input (or just map per batch?... sounds less efficient though)

Reply:
give option to encode all query data cells as same batch (maybe when not specified) -> default :)

Reply:
Looks like, scvi-tools always [assumes](https://docs.scvi-tools.org/en/stable/tutorials/notebooks/query_hlca_knn.html#Download-query-data) that batch is stored under "dataset" key
<img width="755" alt="image" src="https://user-images.githubusercontent.com/35199218/203361824-3edc1230-2a9a-4486-84d1-daedb90a612e.png">

Reply:
I added input parameter to set such key if it doesn't exist. But we should also store batch variable under this key (or change scvi-tools code)